test/new-e2e/tests/agent-shared-components/hostname/imdsv2_transition_common_test.go

```diff
@@ -0,0 +1,197 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package hostname contains helpers and e2e tests of the hostname resolution
+package hostname
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"encoding/json"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awshost "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/host"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client/agentclient"
+)
+
+type Meta struct {
+	Hostname                 string `json:"hostname"`
+	LegacyResolutionHostname string `json:"legacy-resolution-hostname,omitempty"`
+}
+
+type baseHostnameSuite struct {
+	e2e.BaseSuite[environments.Host]
+
+	hostnameMetadata Meta
+}
+
+func TestBaseHostnameSuite(t *testing.T) {
+	e2e.Run(t, &baseHostnameSuite{}, e2e.WithProvisioner(awshost.ProvisionerNoFakeIntake()))
+}
+
+func requestAgentHostnameMetadataPayload(v *baseHostnameSuite) {
+	v.T().Helper()
+
+	// Get the hostname and legacy-resolution-hostname from the agent status
+	statusArgs := agentclient.WithArgs([]string{"header", "--json"})
+	statusPayload := v.Env().Agent.Client.Status(statusArgs)
+
+	var data map[string]interface{}
+	if err := json.Unmarshal([]byte(statusPayload.Content), &data); err != nil {
+		v.T().Fatal(err)
+	}
+
+	metaMap, ok := data["metadata"].(map[string]interface{})["meta"].(map[string]interface{})
+	if !ok {
+		v.T().Fatal()
+	}
+
+	var hostnameMeta Meta
+	hostnameMeta.Hostname, _ = metaMap["hostname"].(string)
+	hostnameMeta.LegacyResolutionHostname, _ = metaMap["legacy-resolution-hostname"].(string)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Just to check type assertion and value exist in the different maps.
It also requires to add `"github.com/stretchr/testify/require"` in the import section. 
```suggestion
	var data map[string]interface{}
	
	// Check for the existence and type of "metadata"
	metadata, ok := data["metadata"]
	require.True(v.T(), ok, "metadata key should exist")
	metaDataMap, ok := metadata.(map[string]interface{})
	require.True(v.T(), ok, "metadata should be of type map[string]interface{}")
	
	// Check for the existence and type of "meta"
	meta, ok := metaDataMap["meta"]
	require.True(v.T(), ok, "meta key should exist")
	metaMap, ok := meta.(map[string]interface{})
	require.True(v.T(), ok, "meta should be of type map[string]interface{}")
	
	// Extract and check types of specific keys within metaMap
	hostname, ok := metaMap["hostname"].(string)
	require.True(v.T(), ok, "hostname should be of type string")
	hostnameMeta.Hostname = hostname
	
	legacyHostname, ok := metaMap["legacy-resolution-hostname"].(string)
	require.True(v.T(), ok, "legacy-resolution-hostname should be of type string")
	hostnameMeta.LegacyResolutionHostname = legacyHostname
```

Reply:
applied with some small changes